pandas/core/indexes/multi.py

```diff
@@ -2715,9 +2717,14 @@ def difference(self, other):
 
     @Appender(_index_shared_docs['astype'])
     def astype(self, dtype, copy=True):
-        if not is_object_dtype(np.dtype(dtype)):
-            raise TypeError('Setting %s dtype to anything other than object '
-                            'is not supported' % self.__class__)
+        dtype = pandas_dtype(dtype)
+        if is_categorical_dtype(dtype):
+            msg = '> 1 ndim Categorical are not supported at this time'
```

Comment:
test fo this?

Reply:
Wrote a test for it in `test_multi.py`, which overrides the test in `common.py`:

https://github.com/jschendel/pandas/blob/3697d6e49631725a2ecaa91a26f9ae97925b197b/pandas/tests/indexes/test_multi.py#L558-L568